### ISSUE TITLE
doesn't work without native twitter installed

### ISSUE THREAD

**Issue by sahyun1** [chroa/twitter-connect-plugin#2]
like ManifestWebDesign:master, this one doesn't work without native twitter app installed on the device.

I'm using firebase and for interim my solution is to use https://ionicframework.com/docs/native/app-availability/ to check if twitter is installed.

If twitter is not avail, use cordova in-app-brwser
https://firebase.google.com/docs/auth/web/cordova

**Comment by chroa** [comment]
Hmm that is interesting! I have been noticed about this issue and I didn't know how to solve it. I shall look into that! :)

**Comment by c00** [comment]
If you do solve it, you will probably become the main fork of this plugin. I have this issue, and I see people complaining about this everywhere (Ionic forums, SO, comments of tutorials, etc.). 

You'd also receive my eternal gratitude :+1:

**Comment by rolinger** [comment]
Is it even possible to do web based login like you can with FB?  I thought the app had to be installed regardless?

**Comment by Fatimaboet** [comment]
Hello! I have the same problem when the native twitter app is not installed the plugin does not work and generated "failed login session" error. Do you have any solution for this? 
I'd be very gratefull!!

**Comment by JaponicaGR** [comment]
Hi. I have twitter app installed on mu device.
When i try to implement twitter login with ionic a popup opens and say "Authorize MyAppName to use your account" and ask my twitter credentials.
Is that the default behaviour?
Isn't supposed to just ask me to login with my account without the need to fill a form with my twitter credentials?

**Comment by NavKumarV** [comment]
In the twitter app account [  https://apps.twitter.com/ ] where u generated the api keys, put the callback URL as **twittersdk://** and then try to login from your app.

**Comment by kushaldias** [comment]
any solution for this issue?

**Comment by djabif** [comment]
@kushaldias, @NavKumarV did anyone managed to get the Native Login working?

**Comment by c00** [comment]
Yeah, native login works fine. As long as you have the Twitter app installed on your phone.

**Comment by djabif** [comment]
I can't manage to get the Native login working in iOS... it works ok on android. Any hint? thanks!

**Comment by c00** [comment]
Without any more info than "Its not working", not really ;)

But here's a general guideline:

1. Make sure the Twitter app is installed
2. Make sure you are logged in in the twitter app (Apparently also important)
3. After 1 and 2, remove your app fromt he phone and reinstall it (shouldn't be necessary, but it wouldn't hurt either).
4. Check out the xcode output to see if that gives any hints to what's going on.
5. Google any error messages, use SO, the usual.

Good luck!
